Source/SwiftLintFramework/Rules/ModifiersOrderRule.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  ModifiersOrderRule.swift
+//  SwiftLint
+//
+//  Created by Jose Cheyo Jimenez on 05/06/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+private extension Array where Element == String {
+    func sorted(byComparingTo reference: [String]) -> [String] {
+        return self.sorted(by: { (lhs, rhs) -> Bool in
+            guard let left = reference.index(of: lhs), let right = reference.index(of: rhs)
+                else { return false }
+            return left < right
+        })
+    }
+}
+
+public struct ModifiersOrderRule: OptInRule, ConfigurationProviderRule {
+    public init() { }
+    public var configuration = ModifiersOrderConfiguration(beforeACL: ["override"], afterACL: [])
+    public static let description = RuleDescription(
+        identifier: "modifiers_order",
+        name: "Modifiers Order",
+        description: "Modifiers order should be consistent.", kind: RuleKind.style,
+        nonTriggeringExamples: [
+            "@objc \npublic final class MyClass: NSObject {\n" +
+            "private final func myFinal() {}\n" +
+            "weak var myWeak: NSString? = nil\n" +
+            "public static let nnumber = 3 \n }",
+
+            "public final class MyClass {}"
+        ],
+        triggeringExamples: [
+            "@objc \npublic final class MyClass: NSObject {\n" +
+            "final private func myFinal() {}\n}",
+
+            "@objc \nfinal public class MyClass: NSObject {}\n",
+
+            "final public class MyClass {}\n",
+
+            "class MyClass {" +
+            "weak internal var myWeak: NSString? = nil\n}",
+
+            "class MyClass {" +
+            "static public let nnumber = 3 \n }"
+        ]
+    )
+
+    private func extractContinuousSections(from tokens: [SyntaxToken], in file: File) -> [[SyntaxToken]] {
```

Comment:
If I can remember right this will give you have arrays where builtins, keywords happen. As in `weak override` perhaps is one continuous sub array of same type of keywords.